db/migrate/20240109180000_fix_unread_comments_inconsistencies.rb

```diff
@@ -0,0 +1,55 @@
+# Fixes the unread_comments flag for all users. Unintended behavior was
+# introduced in pull request #515. Migration introduced in #587.
+# Behavior fixed in #585.
+#
+# This migration is generally *not* idempotent since users might have interacted
+# with the website since the migration was run and thus they will probably have
+# different unread comments flags as the ones at the time of the migration.
+#
+# This migration is not reversible as we don't store the previous state of
+# the unread_comments flag.
+class FixUnreadCommentsInconsistencies < ActiveRecord::Migration[7.0]
+  def change
+    num_total_users = 0
+    num_fixed_users = 0
+
+    User.find_each do |user|
```

Comment:
I think you can just write
```ruby
User.find_each.with_index do |user, num_total_users| 
```
and then you can omit the `num_total_users = 0` and `num_total_users +=1` (did not check it though, just from memory). Detailed report to follow tomorrow or the day after.

Reply:
I don't know why I counted manually in the first place as I could just use `array.length` to get the total number ;) Fixed this.